internal/output/table.go

```diff
@@ -212,64 +209,63 @@ type tbInnerResponse struct {
 }
 
-func tableBuilderInner(vulnResult *models.VulnerabilityResults, calledVulns bool, unimportantVulns bool) []tbInnerResponse {
+func tableBuilderInner(result Result, vulnAnalysisType VulnAnalysisType) []tbInnerResponse {
 	allOutputRows := []tbInnerResponse{}
-	workingDir := mustGetWorkingDirectory()
+	// workingDir := mustGetWorkingDirectory()
 
-	for _, sourceRes := range vulnResult.Results {
-		for _, pkg := range sourceRes.Packages {
-			source := sourceRes.Source
-			sourcePath, err := filepath.Rel(workingDir, source.Path)
-			if err == nil { // Simplify the path if possible
-				source.Path = sourcePath
-			}
+	for _, eco := range result.Ecosystems {
+		for _, source := range eco.Sources {
+			for _, pkg := range source.Packages {
+				var everything []VulnResult
 
-			// Ensure that groups are sorted consistently using the first ID in each group
-			slices.SortFunc(pkg.Groups, func(a, b models.GroupInfo) int {
-				return identifiers.IDSortFunc(a.IDs[0], b.IDs[0])
-			})
+				everything = append(everything, pkg.RegularVulns...)
+				everything = append(everything, pkg.HiddenVulns...)
 
-			// Merge groups into the same row
-			for _, group := range pkg.Groups {
-				if !(group.IsCalled() == calledVulns && group.IsGroupUnimportant() == unimportantVulns) {
-					continue
-				}
+				for _, vuln := range everything {
+					outputRow := table.Row{}
+					shouldMerge := false
 
-				outputRow := table.Row{}
-				shouldMerge := false
+					var links []string
+
+					if vuln.VulnAnalysisType != vulnAnalysisType {
+						continue
+					}
 
-				var links []string
+					for _, id := range vuln.GroupIDs {
+						links = append(links, OSVBaseVulnerabilityURL+text.Bold.Sprintf("%s", id))
 
-				for _, vuln := range group.IDs {
-					links = append(links, OSVBaseVulnerabilityURL+text.Bold.Sprintf("%s", vuln))
+						// For container scanning results, if there is a DSA, then skip printing its sub-CVEs.
+						if strings.Split(id, "-")[0] == "DSA" {
+							break
+						}
+					}
 
-					// For container scanning results, if there is a DSA, then skip printing its sub-CVEs.
-					if strings.Split(vuln, "-")[0] == "DSA" {
-						break
+					outputRow = append(outputRow, strings.Join(links, "\n"))
+
+					// todo: this is just to make the snapshots pass without change
+					if vuln.SeverityScore == "N/A" {
+						outputRow = append(outputRow, "")
+					} else {
+						outputRow = append(outputRow, vuln.SeverityScore)
 					}
-				}
 
-				outputRow = append(outputRow, strings.Join(links, "\n"))
-				outputRow = append(outputRow, group.MaxSeverity)
+					outputRow = append(outputRow, eco.Name)
+
+					name := pkg.Name
 
-				if pkg.Package.Ecosystem == "" && pkg.Package.Commit != "" {
-					pkgCommitStr := results.PkgToString(pkg.Package)
-					outputRow = append(outputRow, "GIT", pkgCommitStr, pkgCommitStr)
-					shouldMerge = true
-				} else {
-					name := pkg.Package.Name
 					// TODO(#1646): Migrate this earlier to the result struct directly
```

Comment:
I'm not sure if I've actioned this as a happy accident, or if it's about moving the whole condition (and thus the need for my new `DepGroups` field) out earlier - if it is the latter though, is it meaning we append the dev marker elsewhere? that feels a bit weird as we'd be changing the "name" of the package and what if we need that for other logic...?

Reply:
I think we can just modify package.name earlier rather than adding DepGroups. This Result struct is only for displaying the result and is being used for other formats as well. I think it would be better that package.name matches the final output, and it also avoids extra handling in other formats